packages/zoe/src/contracts/otcDeskReputation.js

```diff
@@ -0,0 +1,142 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { assert, details } from '@agoric/assert';
+import { trade } from '../contractSupport';
+
+import '../../exported';
+
+/**
+ * This OTC Desk contract does not assume that the underlying assets
+ * have been escrowed. Rather, a reputation mechanism
+ * is used to inform potential users of how often the quote is
+ * accurate, and how often the trade fails.
+ *
+ * @type {ContractStartFn}
+ */
+const start = zcf => {
+  const { zcfSeat: marketMakerSeat } = zcf.makeEmptySeatKit();
+
+  let numTradesAttempted = 0;
+  let numTradesDefected = 0;
+
+  const makeQuote = (price, assets, timerAuthority, deadline) => {
+    let quoteActive = true;
+    E(timerAuthority).setWakeup(
+      deadline,
+      harden({ wake: () => (quoteActive = false) }),
+    );
+    const acceptQuote = seat => {
+      assert(quoteActive, details`The quote was no longer active`);
```

Comment:
Does the quote automatically go inactive when the deadline passes?  (so that expired quotes don't count against the totals)

Reply:
Yup, it automatically goes in-active.